pints/tests/test_opt_cmaes_bare.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python3
+#
+# Tests the basic methods of the bare-bones CMAES optimiser.
+#
+# This file is part of PINTS (https://github.com/pints-team/pints/) which is
+# released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+import unittest
+import numpy as np
+
+import pints
+import pints.toy
+
+from shared import CircularBoundaries
+
+debug = False
+method = pints.BareCMAES
+
+# Consistent unit testing in Python 2 and 3
+try:
+    unittest.TestCase.assertRaisesRegex
+except AttributeError:
+    unittest.TestCase.assertRaisesRegex = unittest.TestCase.assertRaisesRegexp
+
+
+class TestBareCMAES(unittest.TestCase):
+    """
+    Tests the basic methods of the BareCMAES optimiser.
+    """
+    def setUp(self):
+        """ Called before every test """
+        np.random.seed(3)
+
+    def problem(self):
+        """ Returns a test problem, starting point, sigma, and boundaries. """
+        r = pints.toy.ParabolicError()
+        x = [0.1, 0.1]
+        s = 0.1
+        b = pints.RectangularBoundaries([-1, -1], [1, 1])
+        return r, x, s, b
+
+    def test_unbounded(self):
+        # Runs an optimisation without boundaries.
+        r, x, s, b = self.problem()
+        opt = pints.OptimisationController(r, x, method=method)
+        opt.set_threshold(1e-3)
+        opt.set_log_to_screen(debug)
+        found_parameters, found_solution = opt.run()
+        self.assertTrue(found_solution < 1e-3)
+
+    def test_bounded(self):
+        # Runs an optimisation with boundaries.
+        r, x, s, b = self.problem()
+
+        # Rectangular boundaries
+        b = pints.RectangularBoundaries([-1, -1], [1, 1])
```

Comment:
Why define boundaries again, when `b` has been defined as those boundaries already in `problem`?

Reply:
Ok, probably to make the difference to the `CircularBoundaries` more explicit.